Assets/Scripts/StarInfoPanel.cs

```diff
@@ -36,14 +22,31 @@ public void setEnabled(bool enabled)
         }
     }
 
-    public void UpdateStarInfoPanel(string XByerFlamsteed, string magnitude, string constellation)
+    public void UpdateStarInfoPanel(Star starData)
     {
         if (isEnabled && !gameObject.activeSelf)
         {
             gameObject.SetActive(true);
         }
-        starXByerFlamsteedText.text = "ID: " + XByerFlamsteed;
-        starMagText.text = "Mag: " + magnitude;
-        starConstellationText.text = "Const: " + constellation;
+        string description = "";
```

Comment:
Replace this with a StringBuilder - it's way more efficient for concatenating strings, plus you can AddLine a string which automatically adds the line break

Reply:
Fixed.  Good thinking!